tool/resources/org/antlr/v4/tool/templates/codegen/JavaScript/JavaScript.stg

```diff
@@ -91,7 +91,6 @@ exports.<file.grammarName>Listener = <file.grammarName>Listener;
 VisitorFile(file, header) ::= <<
 <fileHeader(file.grammarFileName, file.ANTLRVersion)>
 var antlr4 = require('antlr4/index');
-<header>
 
```

Comment:
The generated header looks like this:
```javascript
var TVisitor = require('./TVisitor').TVisitor;
```
which leads to `TVisitor` being `undefined` even after
```javascript
function TVisitor() {
    …
}
```
Consequently trying to access `TVisitor.prototype` yields an exception.

If you compare the VisitorFile template with the ListenerFile template you'll see, that the latter one lacks the `<header>` as well.